### DISCUSSION TITLE
Index is duplicated for an array

### DISCUSSION THREAD

**Discussion by abhishekrongali** [schibsted/jslt#315] [General]
Can you please help me with how to avoid duplicate of index when Iterating through array list ?

**Example:**
**array** -> [{"value" : "a"},  {"value" : "b"},  {"value" : "a"}]

I used **index-of(array, value)** then I got below index values

 {"value" : "a"} -> **index 0**
  {"value" : "b"} -> **index 1**
  {"value" : "a"} -> **index 0**        // **(Expecting index value should be 2)**

Even values are same, it should have incremental index!

Thanks in advance

**Comment by larsga** [comment]
The function gives you the array index of the first value in the array that's the same as the value you gave it. 

The first and third invocations of the function are the same, so of course they must give the same result.

I don't know what to suggest here, as I can't tell what you're trying to do.

**Reply by abhishekrongali** [reply]
Basically I have an arraylist of different json records, I want to loop through the arraylist and I need index value to concatenate to a value! 

**Example :**
for(array){
 "count" : index-of(array, .)
}
For example I have list of 10 records, when I loop through it, index should start as 0,1,2....till 9.
It should not repeat the index even few records are same   

If any other approach available please suggest!
Thank you

**Reply by larsga** [reply]
Have you tried `zip-with-index`? That is:

```
  for(zip-with-index(array)) {
    "count" : .index
  }
```

**Reply by abhishekrongali** [reply]
Thank you so much, works like a charm..

**Comment by catull** [comment]
Perhaps he wishes to have a different function.

Something like `find-nth-index-of (array, occurence, value)`, and he would call it analogously:

```
{ "occurence": find-nth-index-of (array, 2, { "value": "a" }) }
```

.... with negative occurence indices specify from the tail rather than from head.

Just an idea.

**Reply by larsga** [reply]
It's possible. Or first occurrence after index no `x`. But more likely the real need is something completely different. We need more info to tell.
